cpu/cortexm_common/vectors_cortexm.c

```diff
@@ -101,11 +109,26 @@ void reset_handler_default(void)
     for (dst = &_srelocate; dst < &_erelocate; ) {
         *(dst++) = *(src++);
     }
+
+    /* FIXME: ifdef this!!!! */
+    /* load low-power data section. */
+    for (dst = _sbackup_data, src = _sbackup_data_load;
```

Comment:
If we do this unconditionally it defeats the purpose of the backup RAM.
Luckily there is usually an arch-specific register that tells us whether we just woke from Standby/Deep Sleep. 

I'd suggest we have something like `bool cpu_woke_from_standby(bool clear_flag);` 

On stm32:
```
bool cpu_woke_from_standby(bool clear_flag) {
  bool ret = PWR->CSR & PWR_FLAG_SB;
  if (clear_flag)
     PWR->CR |= PWR_FLAG_SB << 2;
   return ret;
}
```

On saml21 & same54:
```
bool cpu_woke_from_standby(bool clear_flag) {
  return RSTC->RCAUSE.bit.BACKUP; // TODO: no way to clear this?
}
```

Reply:
mmm, I see. I didn't have the complete picture. So, what you are saying is that when the MCU wakes up from Standby/Deep Sleep.the reset vector is invoked again, right? Then yes we need a check, and it also makes me wonder if `.backup.noinit` make any sense (I guess no).

This _totally_ needs a test!

> bool cpu_woke_from_hibernate(bool clear_flag);

mind doing a PR for that function only?

Reply:
Yes, in the deepest sleep mode no state is kept¹ - the system is essentially off and usually can be woken by a GPIO (if previously configured) or RTC.

[1] unless configured otherwise